backend/src/libs/packages/server-application/base-server-application.ts

```diff
@@ -136,46 +150,27 @@ class BaseServerApplication implements ServerApplication {
   private initErrorHandler(): void {
     this.app.setErrorHandler(
       (error: FastifyError | ValidationError, _request, reply) => {
-        if ('isJoi' in error) {
-          this.logger.error(`[Validation Error]: ${error.message}`);
-
-          error.details.forEach((detail) => {
-            this.logger.error(
-              `[${detail.path.toString()}] — ${detail.message}`,
-            );
-          });
-
-          const response: ServerValidationErrorResponse = {
-            errorType: ServerErrorType.VALIDATION,
-            message: error.message,
-            details: error.details.map((detail) => ({
-              path: detail.path,
-              message: detail.message,
-            })),
-          };
-
-          return reply.status(HTTPCode.UNPROCESSED_ENTITY).send(response);
-        }
+        const { status, response } = createErrorResponse(error);
 
-        if (error instanceof HTTPError) {
-          this.logger.error(`[HTTP Error]: ${error.status} – ${error.message}`);
-
-          const response: ServerCommonErrorResponse = {
-            errorType: ServerErrorType.COMMON,
-            message: error.message,
-          };
-
-          return reply.status(error.status).send(response);
+        if ('isJoi' in error) {
```

Comment:
Actually this layer doesn't know about error origin, so the result of createErrorResponse should contain the needed data to simplify pass to logger without any error condition(only one can be for example if (response as ServerValidationErrorResponse).details is an array)

Reply:
Completely right

Reply:
just trying to imagine future suggestions: should I create enum for error names ([Auth Error])?

Reply:
No, just a mapper of error name and result string